### ISSUE TITLE
SIGABRT "memory allocation of 8 Gibibyte failed" for simplest README.md

### ISSUE THREAD

**Issue by phip1611** [foxfriends/paper-terminal#5]
I'm using `paper@0.4.0` on Linux 4.19 with Debian 10. I want to open the simplest Markdown file

**README.md**
```markdown
a
```

It fails with `"memory allocation of 8589934592 bytes failed"`  (8 Gibibyte)

**strace: (`$ strace paper README.md`)**
```
execve("/home/phip1611/.cargo/bin/paper", ["paper", "README.md"], 0x7ffd43017b48 /* 25 vars */) = 0
brk(NULL)                               = 0x55fb69285000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=25705, ...}) = 0
mmap(NULL, 25705, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fb0e5138000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libgcc_s.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\3402\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=100712, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fb0e5136000
mmap(NULL, 103472, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fb0e511c000
mprotect(0x7fb0e511f000, 86016, PROT_NONE) = 0
mmap(0x7fb0e511f000, 69632, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x3000) = 0x7fb0e511f000
mmap(0x7fb0e5130000, 12288, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x14000) = 0x7fb0e5130000
mmap(0x7fb0e5134000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x17000) = 0x7fb0e5134000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/librt.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\260#\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=35808, ...}) = 0
mmap(NULL, 39904, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fb0e5112000
mmap(0x7fb0e5114000, 16384, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7fb0e5114000
mmap(0x7fb0e5118000, 8192, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0x7fb0e5118000
mmap(0x7fb0e511a000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x7000) = 0x7fb0e511a000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0@l\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=146968, ...}) = 0
mmap(NULL, 132288, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fb0e50f1000
mmap(0x7fb0e50f7000, 61440, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6000) = 0x7fb0e50f7000
mmap(0x7fb0e5106000, 24576, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x15000) = 0x7fb0e5106000
mmap(0x7fb0e510c000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1a000) = 0x7fb0e510c000
mmap(0x7fb0e510e000, 13504, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fb0e510e000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0p\322\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=1579448, ...}) = 0
mmap(NULL, 1581384, PROT_READ, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fb0e4f6e000
mmap(0x7fb0e4f7b000, 651264, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xd000) = 0x7fb0e4f7b000
mmap(0x7fb0e501a000, 872448, PROT_READ, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xac000) = 0x7fb0e501a000
mmap(0x7fb0e50ef000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x180000) = 0x7fb0e50ef000
close(3)                                = 0
openat(AT_FDCWD, "/lib/x86_64-linux-gnu/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
)      = 14
getcwd("/tmp", 512)                     = 5
mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7fb0e4ba2000
mprotect(0x7fb0e4ba3000, 2097152, PROT_READ|PROT_WRITE) = 0
clone(child_stack=0x7fb0e4da1e30, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tidptr=0x7fb0e4da29d0, tls=0x7fb0e4da2700, child_tidptr=0x7fb0e4da29d0) = 13177
ioctl(0, TCGETS, {B9600 opost isig icanon echo ...}) = 0
epoll_create1(EPOLL_CLOEXEC)            = 10
epoll_ctl(10, EPOLL_CTL_ADD, 0, {EPOLLIN|EPOLLRDHUP|EPOLLET, {u32=0, u64=0}}) = 0
socketpair(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0, [11, 12]) = 0
rt_sigaction(SIGWINCH, NULL, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
rt_sigaction(SIGWINCH, {sa_handler=0x55fb67330790, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART|SA_SIGINFO|SA_NOCLDSTOP, sa_restorer=0x7fb0e5103730}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
epoll_ctl(10, EPOLL_CTL_ADD, 11, {EPOLLIN|EPOLLRDHUP|EPOLLET, {u32=1, u64=1}}) = 0
epoll_wait(10, [], 3, 0)                = 0
openat(AT_FDCWD, "/tmp/README.md", O_RDONLY|O_CLOEXEC) = 13
statx(0, NULL, AT_STATX_SYNC_AS_STAT, STATX_ALL, NULL) = -1 EFAULT (Bad address)
statx(13, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2, ...}) = 0
read(13, "a\n", 32)                     = 2
read(13, "", 30)                        = 0
close(13)                               = 0
mmap(NULL, 135168, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fb0e4b78000
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=13178, si_uid=1000, si_status=127, si_utime=0, si_stime=0} ---
mremap(0x7fb0e4b78000, 135168, 266240, MREMAP_MAYMOVE) = 0x7fb0e4b37000
mremap(0x7fb0e4b37000, 266240, 528384, MREMAP_MAYMOVE) = 0x7fb0e4ab6000
mremap(0x7fb0e4ab6000, 528384, 1052672, MREMAP_MAYMOVE) = 0x7fb0e49b5000
mremap(0x7fb0e49b5000, 1052672, 2101248, MREMAP_MAYMOVE) = 0x7fb0e47b4000
mremap(0x7fb0e47b4000, 2101248, 4198400, MREMAP_MAYMOVE) = 0x7fb0e43b3000
mremap(0x7fb0e43b3000, 4198400, 8392704, MREMAP_MAYMOVE) = 0x7fb0df7ff000
mremap(0x7fb0df7ff000, 8392704, 16781312, MREMAP_MAYMOVE) = 0x7fb0de7fe000
mremap(0x7fb0de7fe000, 16781312, 33558528, MREMAP_MAYMOVE) = 0x7fb0dc7fd000
mremap(0x7fb0dc7fd000, 33558528, 67112960, MREMAP_MAYMOVE) = 0x7fb0d87fc000
mremap(0x7fb0d87fc000, 67112960, 134221824, MREMAP_MAYMOVE) = 0x7fb0d07fb000
mremap(0x7fb0d07fb000, 134221824, 268439552, MREMAP_MAYMOVE) = 0x7fb0c07fa000
mremap(0x7fb0c07fa000, 268439552, 536875008, MREMAP_MAYMOVE) = 0x7fb0a07f9000
mremap(0x7fb0a07f9000, 536875008, 1073745920, MREMAP_MAYMOVE) = 0x7fb0607f8000
mremap(0x7fb0607f8000, 1073745920, 2147487744, MREMAP_MAYMOVE) = 0x7fafe07f7000
mremap(0x7fafe07f7000, 2147487744, 4294971392, MREMAP_MAYMOVE) = 0x7faee07f6000
mremap(0x7faee07f6000, 4294971392, 8589938688, MREMAP_MAYMOVE) = -1 ENOMEM (Cannot allocate memory)
mmap(NULL, 8589938688, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
brk(0x55fd692ac000)                     = 0x55fb692a6000
mmap(NULL, 8590069760, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
mmap(NULL, 8589938688, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
write(2, "memory allocation of ", 21memory allocation of )   = 21
write(2, "8589934592", 108589934592)              = 10
write(2, " bytes failed\n", 14 bytes failed
)         = 14
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 13176
gettid()                                = 13176
tgkill(13176, 13176, SIGABRT)           = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=13176, si_uid=1000} ---
+++ killed by SIGABRT +++
[1]    13173 abort      strace paper README.md
```

Works fine on other plattforms.

**Comment by foxfriends** [comment]
I believe this has been fixed since then, try again on the latest (v2.1.2). If it comes up again with the newest version feel free to reopen.

**Comment by phip1611** [comment]
Hi, I'm sorry! I mixed it up.. because the binary name is "paper", I wanted to update it by running "cargo install paper", which is in fact a totally different project, which caused the error.. "paper-terminal" is running fine. Sorry for the circumstances!
